### PR TITLE
Support dump hash when skip clone icu on Darwin

### PR DESCRIPTION
**Pull request by sunbohong**
```
➜  swift-source ./swift/utils/update-checkout --dump-hashes            
Traceback (most recent call last):
File "./swift/utils/update-checkout", line 15, in <module>
update_checkout.main()
File "/Volumes/sunny/git-code/swift-collection/swift-source/swift/utils/update_checkout/update_checkout/update_checkout.py", line 511, in main
dump_repo_hashes(config)
File "/Volumes/sunny/git-code/swift-collection/swift-source/swift/utils/update_checkout/update_checkout/update_checkout.py", line 322, in dump_repo_hashes
echo=False):
File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/contextlib.py", line 17, in __enter__
return self.gen.next()
File "/Volumes/sunny/git-code/swift-collection/swift-source/swift/utils/swift_build_support/swift_build_support/shell.py", line 157, in pushd
os.chdir(path)
OSError: [Errno 2] No such file or directory: '/Volumes/sunny/git-code/swift-collection/swift-source/icu'
```

 

@swift-ci Please smoke test

### CODE REVIEW

File: utils/update_checkout/update_checkout/update_checkout.py [hunk 1]
```diff
@@ -317,12 +317,17 @@ def dump_repo_hashes(config):
...
+        path = os.path.join(SWIFT_SOURCE_ROOT, repo_name)
+        if os.path.isdir(path):
+            with shell.pushd(path,
+                            dry_run=False,
+                            echo=False):
+                h = shell.capture(["git", "rev-parse", "HEAD"],
+                                echo=False).strip()
+                repos[repo_name] = str(h)
+        if not os.path.isdir(path):
+            repos[repo_name] = "dir not exist"
```
Comment (gottesmm) [inline utils/update_checkout/update_checkout/update_checkout.py]: Can you change this to do an early exit/continue? I.e.:

```
path = os.path.join(SWIFT_SOURCE_ROOT, repo_name)
if not os.path.isdir(path):
    repos[repo_name] = "not cloned"
    continue

with shell.pushd(path,
	         dry_run=False,
	         echo=False):
    h = shell.capture(["git", "rev-parse", "HEAD"],
	              echo=False).strip()
```

**Comment by futurejones** [comment]
Can one of the admins verify this patch?

**Comment by shahmishal** [comment]
Please update the base branch to `main` by Oct 5th otherwise the pull request will be closed automatically. 

* How to change the base branch: ([Link](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/changing-the-base-branch-of-a-pull-request))
* More detail about the branch update: ([Link](https://forums.swift.org/t/updating-branch-names/40412/))

**Comment by sunbohong** [comment]
https://github.com/apple/swift/commit/f5963923de41d011c25f6bc9e2badc3bb92eae7b also fixed this issue

**Pull request closed without merging**